### ISSUE TITLE
[Issue]: Jellyfin thinks all my videos are WEBMs and tries either re-encoding or remuxing.

### ISSUE THREAD

**Issue by JustAHippo** [jellyfin/jellyfin#8353]
### Please describe your bug

Example:
You're in an app that supports HEVC but does NOT support the MKV format. Jellyfin checks if the client can support the codec. If it can, it decides to check if the file format is supported. If it is, the video plays fine.

The issue is that if the file format isn't supported, rather than remuxing, it decides to fully transcode the video. This is horrible for optimization considering remuxing without re-encoding would save a LOT more resources. I would really like to see this implemented

(If the word remuxing is new, it means converting file format without transcoding to another format)

### Jellyfin Version

10.8.0

### Environment

```markdown
- OS: Raspberry Pi OS
- Virtualization: none
- Clients: Web Client
- Browser: Chromium 106.0.5211.0 patched with H265 support
- FFmpeg Version:
- Playback Method:
- Hardware Acceleration: V4L2
- Plugins: none
- Reverse Proxy: none set up
- Base URL: raspberrypi.local(on local network)
- Networking: local device
- Storage: Seagate Portable
```

### Please attach any screenshots here

![image](https://user-images.githubusercontent.com/82006314/188253180-9bce157c-76e7-4ee4-8f66-d7d66afb47c2.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by JustAHippo** [comment]
Since this is an issue within how the code was written, I don't find it appropriate to link information about my system because this is a global issue

**Comment by crobibero** [comment]
I dont understand the bug youve posted, Jellyfin already supports remuxing files.

It is up to the client to provide the support matrix. Please provide concrete examples.

**Comment by JustAHippo** [comment]
I've attached more context to this issue. On an unpatched chrome version, the image would say
"The codec is not supported"
However on this patched version only the container seems to be the issue as presented. This means it tried fully re-encoding just because of the container type.

**Comment by JustAHippo** [comment]
I'm sorry I hadn't realized this before, the container is Webm, file type MKV and codec HEVC

**Comment by JustAHippo** [comment]
Upon further research the container literally cannot be WEBM or ffmpeg would commit sudoku so we have a problem here.

**Comment by JustAHippo** [comment]
Sorry for the third post, everything is detected as a webm when it is clearly matroska(mkv) file. This is another issue of its own and i will change the name of the bug accordingly

**Comment by JustAHippo** [comment]
closing this since the issue is completely out of date in what i was trying to convey and I will make a new one with updated info

**Comment by Shadowghost** [comment]
Just as a side note:
* The transcoding reason might only be one of multiple
* Your original video is 10 bit SDR which most clients don't support, so the video requires transcoding

**Comment by kubedzero** [comment]
I have the same issue. 

I'm directly comparing Emby vs Jellyfin, and Emby is able to direct play an .MKV file with HEVC while Jellyfin Nerd Stats reports "container is not supported" in the "reason for transcoding." I'm using Chrome 109 on Windows.

I see that it might be recognizing the container as webm, but I'm not positive

On the Jellyfin Codec Support page https://jellyfin.org/docs/general/clients/codec-support/ I see Chrome does not seem to fully support HEVC H.265 content. Furthermore, there seems to be a limitation where "HEVC is only supported in MP4, M4V, and MOV containers." As far as I can tell, Emby direct plays all these files while Jellyfin does not.

I'm hoping to get Direct Play parity with Emby.

Jellyfin Nerd Stats:
![Nerd Stats](https://user-images.githubusercontent.com/51148401/215563508-7617076b-b9a7-4dab-9cae-c12b22e88d9b.png)

Jellyfin Media Info:
![Media Info](https://user-images.githubusercontent.com/51148401/215563568-d60af0c6-1319-4553-8091-1bbddcf70b47.png)

Emby Nerd Stats: 
![Emby](https://user-images.githubusercontent.com/51148401/215563680-cca98970-ffe7-4aab-9256-bc979bbcdf27.png)

**Comment by weddingmm** [comment]
Hi,
Same issue here. I tried to open a video in a Matroska/WebM container, encoded with H265 + multiple profiles and jellyfin is always transcoding. 

"The container is not supported" doesn't make sense IMO. I'm using chrome 109 that should definitely support this container, and the codec as well.

I tried to stream a HEVC Main/WebM video with chrome and is working just fine. 

![image](https://user-images.githubusercontent.com/86187961/217233514-4938a0b3-f10b-42e2-9df6-dd78f1435ccf.png)

**Comment by chaplin89** [comment]
Hi,

I've the same problem here. I tried many different combinations of HEVC profiles + containers, seems like I'm not able to stream a video without transcoding.
Depending on the file, the reason can be "Container not supported", "Video codec not supported", or "Audio codec not supported". None of these explanations make sense IMO as I'm using Chrome 109 that should support both the container and the codec.

I tried a sample file from this website http://h265.webmfiles.org/
In "Embed H265" on the same website I'm able to play a video just find in chrome 109, but when I add a video to the jellyfin library it's always being transcoded.

Went through all the options in the setting, I wasn't able to find a single option that could affect this behavior.

I also tried with an android player, same result.
